tests-ng/test_history.py

```diff
@@ -0,0 +1,58 @@
+import re
+from pathlib import Path
+
+import pytest
+
+CONFIG_FILE = "/etc/profile.d/50-nohistory.sh"
+
+
+@pytest.mark.feature("server", reason="needs server feature")
+def test_history_profile_d_file_exists():
+    assert Path(CONFIG_FILE).exists()
+
+
+@pytest.mark.feature("server", reason="needs server feature")
+def test_history_profile_d_contains_required_configuration():
+    config_contents = Path(CONFIG_FILE).read_text()
+
+    patterns = [
+        re.compile(
+            r"""
+            ^\s*    # any whitespace in the beginning of the line is allowed
+            (?!\#)  # fail the search if a comment character is found
+            HISTFILE=/dev/null
```

Comment:
I would prefer to abstract those regexes into a utility function like `has_line(contents, 'HISTFILE=/dev/null')`, so the whole test would be more expressive

```python
assert has_line(contents, 'HISTFILE=/dev/null')
assert has_line(contents, 'readonly HISTFILE')
```

etc

Reply:
not even sure if regex is needed here.. sure the line could have whitespaces between the words ans still be valid shell, but maybe this could be done in a more simple way, like splitting the input line on whitespace and checking that all the words appear in that line in the right order